### PR TITLE
chore: add script to adapt os memory

### PR DESCRIPTION
**Pull request by GeorgeHcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by kiner-tang** [comment]
@SocketSecurity ignore npm/core-js@3.34.0

File: set-node-options.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const os = require('os');
+const childProcess = require('child_process');
+
+const totalMemory = Math.floor(os.totalmem() / (1024 * 1024));
+
+let maxOldSpaceSize;
+if (totalMemory <= 8192) {
+  maxOldSpaceSize = 4096;
+} else if (totalMemory <= 16384) {
+  maxOldSpaceSize = 4096;
```
Comment (kiner-tang) [inline set-node-options.js:10]: 为什么这种情况也要固定成 4096 呀？

File: set-node-options.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const os = require('os');
+const childProcess = require('child_process');
+
+const totalMemory = Math.floor(os.totalmem() / (1024 * 1024));
+
+let maxOldSpaceSize;
+if (totalMemory <= 8192) {
```
Comment (yoyo837) [inline set-node-options.js:7]: 这三个if怎么感觉好累赘，不过，只是开发用，倒还好。

File: set-node-options.js (hunk 1 above)
Comment (GeorgeHcc) [reply set-node-options.js:10]: 嗯嗯，大于16G应该不用管了，但是8G的第一次运行没有缓存要开4G才够，后面其实2G就可以了。
或者直接设置最小内存=4096？

File: set-node-options.js (hunk 2 above)
Comment (afc163) [reply set-node-options.js:6]: 改成 TS 文件？

Comment (GeorgeHcc) [reply set-node-options.js:6]: 那放在scripts文件夹下吗，还是根目录呢

Comment (afc163) [reply set-node-options.js:6]: scripts 目录下。

**Comment by CooperHash** [comment]
#47588

**Review by afc163 — APPROVED** [review]

**Review by afc163 — APPROVED** [review]

**Comment by ammirator-administrator** [comment]
@GeorgeHcc @kiner-tang @CooperHash @afc163 
This doesn't fix anything
In last version 5.17.4 the problem is there the library consumes memory as crazy
```
FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
```

Just by running a simple script with TS to lint the types 
```
tsc --noEmit --pretty --p .",
```
Antd takes all the memory, somethign BAD was added to the library it should not consume 2gb of default memory thats crazy

**Comment by ammirator-administrator** [comment]
This happens because of ANTD design library in a separate project I have this project because of antd

**Pull request merged**